webapp/src/components/training/Training.vue

```diff
@@ -38,22 +38,42 @@ import Finished from '@/components/training/Finished.vue'
 import Data from '@/components/data/Data.vue'
 
 const router = useRouter()
+const route = useRoute()
 const trainingStore = useTrainingStore()
 const tasksStore = useTasksStore()
 
 // task ID given by the route
 interface Props { id: TaskID }
 const props = defineProps<Props>()
 
+function setupTrainingStore() {
+  trainingStore.setTask(route.params.id as string) // more reliable than props.id
+  trainingStore.setStep(1)
+}
+// Init the task once the taskStore has been loaded successfully
+// If it is not we redirect to the task list
 const task = computed(() => {
-  const task = tasksStore.tasks.get(props.id)
-  if (task === undefined) {
-    router.replace({ name: 'not-found' })
-    return
+  if (tasksStore.loadedSuccessfully) {
+    return tasksStore.tasks.get(props.id)
+  }
+  // Redirect to the task list if not loaded yet
+  // This happens when refreshing the page, every task are reset when fetched
+  if (route.name !== 'task-list') {
+    router.replace({ name: 'task-list' })
+  }
+  return undefined
+})
+
+// Addresses the case when users enter a url manually
+// Force the training store to synch with the task specified in the url
+watch(() => route.fullPath, () => {
+  if (route.fullPath !== '/list' && (trainingStore.step == 0 || route.params.id !== props.id)) {
```

Comment:
full link look like too much information, and harder to keep track of (don't forget to update the watch target)

```suggestion
  if (route.name !== 'task-list' && (trainingStore.step == 0 || route.params.id !== props.id)) {
```

also, nitpicking on code style: I do like to have flatter functions' body, where `if` are there to "remove obstacles from the happy path". so, if most of the body of a function is wrapped inside an `if`, I would refactor it to rather return with the inverse condition. (arf, hard to explain, here goes some code)
```ts
function f() {
  if (cond) {
    …
  }
}
```
into
```ts
function f() {
  if (!cond) return
  …
}
```

so in our case here
```ts
() => {
  if (route.name === 'task-list') return
  if (trainingStore.step !== 0 && route.params.id === props.id))
    return
  setupTrainingStore()
}
```
that reads a bit better IMO

Reply:
I was actually using `route.name` but it's sometimes undefined, while `route.fullPath` is always defined. Which means I can be on the list page and route.name is sometimes 'task-list' and sometimes 'undefined' which is just calling for future issues.
I don't have a strong opinion on the function flatness, I'm in lining the method!

Reply:
Even seems that using route.fullPath triggers on mount and let's me remove the onMount hook